core/__tests__/actions/records.ts

```diff
@@ -229,6 +230,50 @@ describe("actions/records", () => {
         await specHelper.runAction<RecordsList>("records:list", connection);
       expect(readyProfiles.length).toBe(1);
       expect(readyTotal).toBe(1);
+
+      await RecordProperty.update(
+        {
+          invalidValue: "email",
+          invalidReason: "Bad Email",
+        },
+        {
+          where: {
+            recordId: readyProfiles[0].id,
+          },
+        }
+      );
+
+      connection.params = {
+        csrfToken,
+        state: "invalid",
+      };
+      const { records: invalidRecords, total: invalidTotal } =
+        await specHelper.runAction<RecordsList>("records:list", connection);
+      expect(invalidRecords.length).toBe(1);
+      expect(invalidTotal).toBe(1);
+    });
+
+    test("a writer can find a record with an invalid state, even if the queried record is not invalid", async () => {
```

Comment:
There might be a better way to phrase this but effectively if any recordProperty is invalid for a user and you're looking for a recordProperty on that user that _is not invalid AND you want invalid records_ this is confirming it will show up.